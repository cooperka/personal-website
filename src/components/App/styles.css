.App {
  display: flex;
  flex-direction: column;
  text-align: center;
}

.main-header {
  margin: 2em;
  margin-bottom: 0;
  color: #444;
}

.profile-photo {
  height: 16em;
  border-radius: 8em;
}

.main-title {
  margin: 1rem;
  font-size: 5em;
}

.main-subtitle {
  margin: 1rem;
  font-size: 2.2em;
}

.social-links {
  margin: 2em;
  margin-top: 0;
  padding: 0;
  list-style: none;
}
