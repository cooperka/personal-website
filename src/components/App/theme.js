export default {
  // https://material-ui-next.com/customization/themes/#typography
  typography: {
    // Account for base font-size of 62.5%.
    htmlFontSize: 10,
  },
  props: {
    MuiTypography: {
      variantMapping: {
        h2: 'h1',
        h5: 'h2',
      },
    },
  },
};
