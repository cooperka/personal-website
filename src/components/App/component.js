// @flow

import React, { Component } from 'react';

import profilePhoto from '../../images/profile-2017-web@800.jpg';
import SOCIAL_LINKS from '../../socialLinks';

import SocialLink from '../SocialLink';

import './styles.css';

type Props = {
};

class App extends Component<Props> {

  renderMainHeader() {
    return (
      <div className="main-header">
        <img className="profile-photo material z-depth-1" src={profilePhoto} alt="profile" />
        <div className="main-title">Kevin Cooper</div>
        <div className="main-subtitle">Software Engineer</div>
        <div className="main-subtitle">I run <a href="https://lsrfocus.com">LSR</a>, providing freelance software development and strategic planning.</div>
      </div>
    );
  }

  renderSocialLinks() {
    return (
      <ul className="social-links">
        {SOCIAL_LINKS.map(
          (data, index) =>
            // eslint-disable-next-line react/no-array-index-key
            <SocialLink key={index} name={data.name} url={data.url} icon={data.icon} iconHover={data.iconHover} />,
        )}
      </ul>
    );
  }

  renderFooter() {
    return (
      <div className="footer">
        <div>&copy; 2018 Kevin Cooper</div>
      </div>
    );
  }

  render() {
    return (
      <div className="App">
        {this.renderMainHeader()}
        {this.renderSocialLinks()}
        {this.renderFooter()}
      </div>
    );
  }

}

export default App;
