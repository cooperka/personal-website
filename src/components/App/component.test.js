// @flow

import React from 'react';
import renderer from 'react-test-renderer';

import App from './component';

jest.mock('./MainHeader/component', () => 'MainHeader');
jest.mock('./SocialLinks/component', () => 'SocialLinks');
jest.mock('./ProjectLinks/component', () => 'ProjectLinks');
jest.mock('./Footer/component', () => 'Footer');

it('renders', () => {
  const tree = renderer.create(<App />);
  expect(tree.toJSON()).toMatchSnapshot();
});
