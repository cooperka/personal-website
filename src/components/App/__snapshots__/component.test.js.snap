// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<div
  className="App"
>
  <div
    className="main-header"
  >
    <img
      alt="profile"
      className="profile-photo"
      src="profile-2017-web@800.jpg"
    />
    <div
      className="main-title"
    >
      Kevin Cooper
    </div>
    <div
      className="main-subtitle"
    >
      Software Engineer
    </div>
  </div>
  <ul
    className="social-links"
  >
    <SocialLink
      icon="Github-black.svg"
      iconHover="Github-color.svg"
      name="GitHub"
      url="https://github.com/cooperka"
    />
    <SocialLink
      icon="LinkedIn-black.svg"
      iconHover="LinkedIn-color.svg"
      name="LinkedIn"
      url="https://www.linkedin.com/in/cooperka92"
    />
    <SocialLink
      icon="Medium-black.svg"
      iconHover="Medium-color.svg"
      name="Medium"
      url="https://medium.com/@cooperka"
    />
    <SocialLink
      icon="Facebook-black.svg"
      iconHover="Facebook-color.svg"
      name="Facebook"
      url="https://www.facebook.com/cooperka"
    />
    <SocialLink
      icon="Twitter-black.svg"
      iconHover="Twitter-color.svg"
      name="Twitter"
      url="https://twitter.com/cooperka92"
    />
    <SocialLink
      icon="Tumblr-black.svg"
      iconHover="Tumblr-color.svg"
      name="Tumblr"
      url="https://mentalmorsels.tumblr.com/"
    />
  </ul>
</div>
`;
